addons/hr/models/hr_employee.py

```diff
@@ -137,6 +142,27 @@ def _read(self, fields):
             record = self.browse(r['id'])
             record._update_cache({k:v for k,v in r.items() if k in fields}, validate=False)
 
+    @api.model
+    def _cron_check_work_permit_validity(self):
+        # Called by a cron
+        # Schedule an activity 1 month before the work permit expires
+        outdated_days = fields.Date.today() + relativedelta(months=+1)
+        nearly_expired_work_permits = self.search([('work_permit_activity_scheduled', '=', False), ('work_permit_expire', '<', outdated_days)])
+        employees_scheduled = self.env['hr.employee']
+        for employee in nearly_expired_work_permits.filtered(lambda employee: employee.parent_id):
+            responsible_user_id = employee.parent_id.user_id.id
+            if responsible_user_id:
+                employees_scheduled |= employee
+                lang = self.env['res.partner'].browse(responsible_user_id).lang
+                formated_date = format_date(employee.env, employee.work_permit_expire, date_format="dd MMMM y", lang_code=lang)
+                employee.activity_schedule(
+                    'mail.mail_activity_data_todo',
+                    note=_('The work permit of %(employee)s expires at %(date)s.',
+                        employee=employee.name,
+                        date=formated_date),
+                    user_id=responsible_user_id)
+        employees_scheduled.work_permit_activity_scheduled = True
```

Comment:
To make it work in batch:
```suggestion
        employees_scheduled.write({'work_permit_activity_scheduled': True})
```